Client/game_sa/CCameraSA.cpp

```diff
@@ -446,8 +446,14 @@ float CCameraSA::GetShakeForce()
 
 void CCameraSA::ShakeCamera(float radius, float x, float y, float z) noexcept
 {
-    DWORD               dwFunc = FUNC_ShakeCam;
-    CCameraSAInterface* cameraInterface = GetInterface();
+    static CCameraSAInterface* cameraInterface = GetInterface();
```

Comment:
You don't need `static` here

Reply:
it can be static so the variable wouldnt be allocated again

Reply:
If this will not be static, the variable won't be allocated at all

Reply:
it will be. It will live throughout the whole program.

Reply:
> If this will **not be static,**

This is what i mean. With out `static` it will be [esp+41C] or something

Reply:
its still being allocated nonetheless. Static just makes one definition for all calls instead of one definition per call.

Reply:
It's won't be allocated, because compiler knows memory position for `internalInterface` at compile time

Reply:
> It won't be allocated, because compiler knows memory position for `internalInterface` at compile time

but the allocation happens at runtime, not compile time. The definition is not evaluated at compile time.
The compiler still needs to allocate memory for the variable. It being static doesnt make much of a difference other than perseverance between calls

Reply:
There is no memory allocation happening. GetInterface() just returns a pointer to the interface. 
Adding static will check TLS to see if camInterface is already initialized every time the function is called.
The function should not be noexcept since it calls a game function.

Reply:
It doesnt matter what GetInterface returns.
The compiler needs to allocated memory on the stack for the `camInterface` variable. No need to do that X amount of times if `&GetInterface()` doesnt change.

Reply:
The return value will be in eax and then moved into ecx to call the function. The compiler is unlikely to move this onto the stack.

Reply:
It needs to be on the stack/somewhere in memory because the class member is modified from that exact variable.

Reply:
It doesn't use an additional memory or stack. Maybe only one additional register will be used, but it's better than an additional branching from static variable

Reply:
> It doesn't use an additional memory or stack. Maybe only one additional register will be used, but it's better than an additional branching from static variable

what do you mean?

Reply:
> Adding static will check TLS to see if camInterface is already initialized every time the function is called.

Reply:
Not adding static will define it each time the function runs.
When static is added, a new label in assembly is created for that variable so it can be referenced without allocating it on the stack each time

Reply:
> Not adding static will define it each time the function runs.

In this case a compiler is clever enough to replace variable allocation with direct memory access by offset. You didn't optimize here anything.